### PR TITLE
Optimize Job Struct and Trait

### PR DESCRIPTION
**Pull request by ZhongliGao**
update

### CODE REVIEW

File: engula/src/job/mod.rs [hunk 1]
```diff
@@ -32,3 +25,8 @@ pub struct CompactionOutput {
     pub input_files: Vec<FileDesc>,
     pub output_file: FileDesc,
 }
+
+#[async_trait]
+pub trait JobRuntime: Send + Sync {
+    async fn spawn(&self, input: CompactionInput) -> Result<CompactionOutput>;
```
Comment (huachaohuang) [inline engula/src/job/mod.rs:31]: Let's rename the function name to `compact` too.

File: engula/src/job/job.proto [hunk 2]
```diff
@@ -3,7 +3,7 @@ package engula.job;
 import "format/format.proto";
 
 service Job {
-  rpc spawn_job(JobRequest) returns (JobResponse) {}
+  rpc compact(JobRequest) returns (JobResponse) {}
```
Comment (huachaohuang) [inline engula/src/job/job.proto:6]: We should rename the request/response name too.

**Review by huachaohuang — CHANGES_REQUESTED** [review]

**Comment by PsiACE** [comment]
hi, @ZhongliGao . You can add something to .gitignore to ignore unnecessary files/dirs

**Comment by ZhongliGao** [comment]
> 你好，@ZhongliGao。您可以向 .gitignore 添加一些内容以忽略不必要的文件/目录
Thanks

**Comment by ZhongliGao** [comment]
> hi, @ZhongliGao . You can add something to .gitignore to ignore unnecessary files/dirs

Thanks

**Pull request closed without merging**